src/value-parser.js

```diff
@@ -335,7 +335,8 @@ function valueParse(parser, metaData, options, callback) {
                     valueMetaData.collation = collation;
                     if (baseType === 0xA5 || baseType === 0xAD || baseType === 0xA7 || baseType === 0xAF || baseType === 0xE7 || baseType === 0xEF) {
                       return readDataLength(parser, valueMetaData.type, {}, null, (maxDataLength) => {
-                        valueMetaData.dataLength = maxDataLength;
+                        // skip the 2-byte max length sent for BIGVARCHRTYPE, BIGCHARTYPE, NVARCHARTYPE, NCHARTYPE, BIGVARBINTYPE and BIGBINARYTYPE types
+                        // and parse based on the length of actual data
```

Comment:
The parsing code seems to assume there is always baseType (1 byte), propBytes (1 byte), precision(1 byte), scale(1 byte), collation(5 bytes), dataLength(2 bytes) and then data. See https://github.com/tediousjs/tedious/blob/master/src/value-parser.js#L333.

Whereas the spec seems to say something else, at least as I read it. The layout is more dynamic. For example, propBytes value is 0 for GUIDTYPE per spec, which means there is no precision or scale and the above code would break. See below:
![image](https://user-images.githubusercontent.com/7972329/27714739-c729948e-5ce7-11e7-9b83-b8d4a5af0847.png)

The code doesn't seem to match anything in the spec. Either that or I'm missing something big time :-)

Perhaps one test could be a Variant with GUIDTYPE contained in it. You can try a few other Variant base types from the table and see if they all work.

And yes, there is some flakiness with Travis test I've seen before.